pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh

```diff
@@ -107,11 +108,27 @@ convertIconTheme() {
             echo "desktopToDarwinBundle: rasterizing svg $in to $out, size $dim" >&2
             rsvg-convert --keep-aspect-ratio --width "$dim" --height "$dim" "$in" --output "$out"
             magick convert -density "$density" -units PixelsPerInch "$out" "$out"
+            convertIfUnsupportedIcon "$out" "$iconSize" "$scale"
         else
             return 1
         fi
     }
 
+    # macOS does not correctly display 16x and 32x png icons on app bundles
+    # they need to be converted to rgb+mask (argb is supported only from macOS 11)
```

Comment:
We’re about to switch unstable to require macOS ≥ 11.3; what would need changing here to use ARGB? (We can keep it like this in the current PR for backport to 24.11).

Reply:
It would only be a matter of:
```
diff --git a/pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh b/pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh
index 4fbef4baa512..6790754f740a 100644
--- a/pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh
+++ b/pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh
@@ -115,15 +115,15 @@ convertIconTheme() {
     }
 
     # macOS does not correctly display 16x and 32x png icons on app bundles
-    # they need to be converted to rgb+mask (argb is supported only from macOS 11)
+    # they need to be converted to argb
     function convertIfUnsupportedIcon() {
         local -r in=$1
         local -r iconSize=$2
         local -r scale=$3
-        local -r out=${in%.png}.rgb
+        local -r out=${in%.png}.argb
 
         if [[ ($scale -eq 1) && ($iconSize -eq 32 || $iconSize -eq 16) ]]; then
-            echo "desktopToDarwinBundle: converting ${iconSize}x icon to rgb" >&2
+            echo "desktopToDarwinBundle: converting ${iconSize}x icon to argb" >&2
             icnsutil convert "$out" "$in"
             rm "$in"
         fi
```

This would be the result:
![relikd github io_icnsutil_html_viewer html](https://github.com/user-attachments/assets/4f46fa41-7ad1-47a6-ad32-9325a655fe4d)